### PR TITLE
feat: add recommendations

### PR DESCRIPTION
**Pull request by rajat-ws**
## Description:

- Added `Recommendations` feature.

## GIF:

22recom

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -1,10 +1,31 @@
...
-  return <TracksContainer />;
+export async function getStaticProps() {
+  const recommendations = await getRecommendations();
+
+  return {
+    props: {
+      recommendations,
+    },
+  };
+}
```
Comment (ihatelactose) [inline pages/index.tsx]: Although it's not a hard requirement, prefer keeping the data fetching at the very bottom of the file.

File: services/root.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { TrackDetail } from "@app/features/trackDetails/api/getTrackDetails";
+import { ApiResponse } from "apisauce";
+import { generateApiClient } from "../utils/apiUtils";
+
+const itunesApi = generateApiClient("itunes");
```
Comment (ihatelactose) [inline services/root.ts]: You can probably create an ENUM for all the types of API clients that are available.

```ts
enum APIClients {
    DEFAULT = 'itunes',
    ITUNES = 'itunes',
}
```

**Review by ihatelactose — APPROVED** [review]
Looks good! Just make the modifications above and this is good to go!

File: pages/index.tsx (hunk 1 above)
Comment (rajat-ws) [reply pages/index.tsx]: Got it.

File: services/root.ts (hunk 2 above)
Comment (rajat-ws) [reply services/root.ts]: Updated accordingly.

**Pull request merged**